training.h

```diff
@@ -25,11 +25,16 @@
 #include <stdlib.h>
 #include <string.h>
 
-
+/* FIXME: This is a giant 558 KB struct which is awful for locality. Many bytes
```

Comment:
We can go beyond float32: fixed16 will likely work for those mountains of clamped log probs. Well, ~140 K is still too big for the average desktop's L1D cache, so I should just stop thinking about this stuff.